Add bivariate - closes #21

neerajt opened the pull request:
A bit of work left to do. Will try to incorporate integration tests with the test data.

neerajt committed:
Add bivariate.ex; Add Covariance

neerajt committed:
fix namespace error on Covariance

crertel commented:
An update on what we have to get done:

* ~Calculate covariance~
* Pearson correlation
* t-statistic
* simple linear regression beta coefficients.

crertel committed:
Fix some compliation errors.

Updates the mixfile to not use function invocation shorthand.
This also notes that one of the params to `pluck_all/3` is unsued.

crertel committed:
Add `pearson_correlation/2`.

neerajt committed:
fix typo

examaples -> examples

crertel committed:
Add `pearson_correlation/2` to Statisaur.

neerajt committed:
Add simple linear regression

Only takes an x and y; returns number of coeffs (2) and b0 and b1

neerajt committed:
Merge with covariance and regression in bivariate

crertel committed:
Fix some namespacing errors on compilation.

There might be a better way of solving this using `alias`, will experiment.

crertel commented:
An update on what we have to get done:

* ~Calculate covariance~
* ~Pearson correlation~
* t-statistic
* simple linear regression beta coefficients.

crertel committed:
Add (untesteD) `pooled_sttdev` function.

This is going to be helpeful when implementing the t-scoring stuff.

crertel committed:
Add test for `pooled_stddev`.

crertel committed:
Add test for t-value.

crertel committed:
Add proper guard clauses to `t_score`.

neerajt committed:
Add pooled_stderr

neerajt committed:
Fix pooled_stderr

oops, was using r syntax!

neerajt committed:
Update t test to use pooled_stderr

neerajt committed:
partially fix pooledstderr

crertel committed:
Fix broken test for `pooled_stderr`.

crertel commented:
We finished it alllll
